CIP-XXXX/README.md

```diff
@@ -0,0 +1,205 @@
+---
+CIP: 92
+Title: First-class errors in Plutus
+Category: Plutus
+Status: Proposed
+Authors:
+    - Las Safin <me@las.rs>
+Implementors:
+    - Las Safin <me@las.rs>
+Discussions:
+    - https://github.com/cardano-foundation/cips/pulls/?
+Created: 2023-03-01
+License: CC-BY-4.0
+---
+
+## Abstract
+
+Plutus is a strictly evaluated language, where users heavily rely on **impure** errors
+to cause transaction failure to rule out invalid cases.
+
+We propose three changes:
+- Make `error` a first-class value you can pass around. Using it does not inherently
+  cause an error.
+- Change condition for script success from not erring to returning `True`.
+- Erase type abstractions and type applications entirely when going from
+  TPLC to UPLC rather than turning them into `delay` and `force`.
+
+UPLC is now **truly pure** wrt. transaction validation.
+This enables another use-case: We can **optimise** it as if it
+were entirely pure. We can turn `const () x` into `()` for any `x`!
+We can turn `expensive1 || expensive2` into `if expensive1 then True else expensive2`,
+and in general do any transformations that we'd do on a pure language without
+regard for side-effects.
+
+Issues fixed / PRs closed by this CIP:
```

Comment:
This list is too much detail for the _Abstract_ and in fact is _Motivation_ so should be moved down to that section.